test-framework/kubernetes-client/src/main/java/io/quarkus/test/kubernetes/client/AbstractKubernetesFixturesTestResource.java

```diff
@@ -0,0 +1,76 @@
+package io.quarkus.test.kubernetes.client;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import io.fabric8.kubernetes.api.builder.Visitor;
+import io.fabric8.kubernetes.api.model.HasMetadata;
+
+public abstract class AbstractKubernetesFixturesTestResource extends
+        AbstractNamespaceCreatingTestResource {
+    private List<HasMetadata> resourceFixtures = Collections.emptyList();
+    private int secondsToWaitForFixturesReadiness;
+    private final LoggingVisitor deleteLogger = new LoggingVisitor("Deleting");
+    private final LoggingVisitor createLogger = new LoggingVisitor("Creating");
+
+    @Override
+    protected Map<String, String> doStart() {
+        if (!resourceFixtures.isEmpty()) {
+            final var resources = client()
+                    .resourceList(resourceFixtures)
+                    .accept(HasMetadata.class, createLogger);
```

Comment:
The `HasMetadata.class` is redundant, since you are passing an instance of visitor.
The type is there in case you need to use lambda expression, in which case the actual type can't be inferred.